src/cvxpylayers/interfaces/mpax_if.py

```diff
@@ -32,103 +34,211 @@ class MPAX_ctx:
+                ),
+                lin_obj_vals_i[self.ctx.c_slice],
+                jax.experimental.sparse.BCSR(
+                    (con_vals_i[self.ctx.A_idxs], *self.ctx.A_structure),
+                    shape=self.ctx.A_shape,
+                ),
+                b_vals,
+                jax.experimental.sparse.BCSR(
+                    (con_vals_i[self.ctx.G_idxs], *self.ctx.G_structure),
+                    shape=self.ctx.G_shape,
+                ),
+                h_vals,
+                self.ctx.l,
+                self.ctx.u,
+            )
+
+            # Solve
+            solution = self.ctx.solver(model)
             return solution.primal_solution, solution.dual_solution
-        solution, fun = jax.vjp(solver, self.model)
-        return *solution, fun
-    
-    def torch_solve(self):
+
+        # Vectorize over batch dimension (axis 1 of parameter arrays)
+        solve_batched = jax.vmap(solve_single_batch, in_axes=(1, 1, 1))
+
+        def batched_solver(quad_vals, lin_vals, con_vals):
+            return solve_batched(quad_vals, lin_vals, con_vals)
+
+        # Compute forward pass and VJP function
+        (primal, dual), vjp_fun = jax.vjp(
+            batched_solver,
+            self.quad_obj_values,
+            self.lin_obj_values,
+            self.con_values,
+        )
+
+        return primal, dual, vjp_fun
+
+    def torch_solve(self, solver_args=None):
         import torch
-        primal, dual, fun = self.jax_solve()
-        return torch.utils.dlpack.from_dlpack(primal), \
-               torch.utils.dlpack.from_dlpack(dual), \
-               fun
 
+        primal, dual, vjp_fun = self.jax_solve()
+        # Convert JAX arrays to PyTorch tensors
+        # jax_solve returns shapes: (batch_size, n) and (batch_size, m)
+        primal_torch = torch.utils.dlpack.from_dlpack(primal)
+        dual_torch = torch.utils.dlpack.from_dlpack(dual)
+        return (
+            primal_torch,
+            dual_torch,
+            vjp_fun,
+        )
 
     def jax_derivative(self, primal, dual, fun):
-        return fun((primal, dual))
+        raise NotImplementedError(
+            "Backward pass is not implemented for MPAX solver. "
+            "Use solver='DIFFCP' for differentiable optimization layers."
+        )
 
-    def torch_derivative(self, primal, dual, fun):
+    def torch_derivative(self, primal, dual, adj_batch):
         import torch
-        quad, lin, con = self.jax_derivative(jnp.array(primal), jnp.array(dual), fun)
+
+        # Squeeze batch dimension (MPAX doesn't support batching, always has batch_size=1)
+        primal_unbatched = primal.squeeze(0) if primal.dim() > 1 else primal
+        dual_unbatched = dual.squeeze(0) if dual.dim() > 1 else dual
+
+        quad, lin, con = self.jax_derivative(
+            jnp.array(primal_unbatched), jnp.array(dual_unbatched), adj_batch
+        )
         return torch.tensor(quad), torch.tensor(lin), torch.tensor(con)
```

Comment:
Shouldn't we use dlpack? I think doing Jax to torch this way raises an error about the read only flag